src/Range.ts

```diff
@@ -5,46 +5,36 @@ export class Range {
      * @param _limits inclusive limits. All numbers outside of it are nor in nor out of the range
      * @param _range range inside of the limits. All ranges are inclusive.
      */
-    constructor(private _limits: [number, number], private _range: Array<[number, number]>) {
+    constructor(private _limits: [number, number], private _range: Array<[number, number]>, private _inverted = false) {
         // blank
     }
 
     in(val: number): boolean {
-        let index = _.sortedIndexBy(this._range, [val, val], item => item[1]);
-        if (index >= this._range.length) {
-            index = this._range.length - 1;
+        if (!this._checkLimits(val)) {
+            return false;
+        } else {
+            let result = false;
+            let index = _.sortedIndexBy(this._range, [val, val], item => item[1]);
+            if (index === this._range.length) {
+                result = false;
```

Comment:
@dryganets By default `result` equals `false` in line 16, does that make sense to change `false` to `false`?